kalite/topic_tools/base.py

```diff
@@ -39,27 +38,20 @@
     logging.warning("Channel {channel} does not exist.".format(channel=settings.CHANNEL))
 
 
-def get_assessment_item_data(request, assessment_item_id=None):
-    try:
-        assessment_item = main_models.AssessmentItem.objects.get(id=assessment_item_id)
-    except main_models.AssessmentItem.DoesNotExist:
-        return None
-    except DatabaseError:
-        return None
-
-    try:
-        item_data = json.loads(assessment_item.item_data, object_hook=json_ascii_decoder)
-        item_data = smart_translate_item_data(item_data)
-        item_data = json.dumps(item_data)
-    except KeyError as e:
-        logging.error("Assessment item did not have the expected key %s. Assessment item: \n %s" % (e, assessment_item))
-
-    #  Expects a dict
-    return {
-        "id": assessment_item.id,
-        "item_data": item_data,
-        "author_names": assessment_item.author_names,
-    }
+def database_exists(channel="khan", language="en", database_path=None):
+    path = database_path or settings.CONTENT_DATABASE_PATH.format(channel=channel, language=language)
+
+    return os.path.exists(path)
+
+def available_content_databases():
+    """
+    Generator to return the channel and language for every content database that exists in the system.
+    """
+    pattern = re.compile("content_([^_]+)_([^_]).sqlite")
```

Comment:
Should it be `content_([^_]+)_([^_]+).sqlite`? [What's there doesn't match `content_en_khan.sqlite`](http://pythex.org/?regex=content_(%5B%5E_%5D%2B)_(%5B%5E_%5D).sqlite&test_string=content_en_khan.sqlite&ignorecase=0&multiline=0&dotall=0&verbose=0)

It's not entirely obvious, what order is the channel & language supposed to be returned in? `("khan", "en")` or `("en", "khan")`? Add to docstring or use named groups, or better yet both.